### PR TITLE
Make usagegraphgenerator module non-interface classes internal

### PR DESCRIPTION
**Pull request by JSAbrahams**
* Make the `companion object` of `BranchStatementFilter` private, as it is not used elsewhere
* Make filters internal, as these are not part of the interface
* Make extension function in `ControlFlowGraphGenerator` private, including the accompanying Exception, as it used only by this class.
* Make `DotGraphRendered` internal, as it is not part of the interface

### CODE REVIEW

File: modules/pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/pipeline/usagegraphgenerator/jimple/filters/BranchStatementFilter.kt [hunk 1]
```diff
@@ -42,13 +42,13 @@ internal class BranchStatementFilter(project: JavaProject) : Filter {
...
     private companion object {
-        private fun isBranchStatement(statement: Unit) = when (statement) {
+        fun isBranchStatement(statement: Unit) = when (statement) {
             is IfStmt -> true
             is SwitchStmt -> true
             else -> false
         }
 
-        private fun getConditionValue(statement: Unit): Value = when (statement) {
+        fun getConditionValue(statement: Unit): Value = when (statement) {
```
Comment (FWDekker) [inline modules/pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/pipeline/usagegraphgenerator/jimple/filters/BranchStatementFilter.kt]: Why were these changed from `private` to `public`?

Comment (JSAbrahams) [reply modules/pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/pipeline/usagegraphgenerator/jimple/filters/BranchStatementFilter.kt]: The companion object is private so the internal functions are no longer accessible.

Comment (FWDekker) [reply modules/pipeline/jimple-library-usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/pipeline/usagegraphgenerator/jimple/filters/BranchStatementFilter.kt]: Ach so!
